src/uu/cksum/src/cksum.rs

```diff
@@ -367,6 +369,14 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
         None
     };
 
+    if algo_name == "bsd" && check {
+        return Err(io::Error::new(
+            io::ErrorKind::InvalidInput,
+            "--check is not supported with --algorithm={bsd,sysv,crc}",
+        )
+        .into());
+    }
+
```

Comment:
So we accept `-c`, callers think that we actually verify the checksum of some files, and then we always exit with 0? This sounds like a terrible idea. Let's either actually support `-c` or not support it, but "accepting the argument and ignoring it" is a security disaster waiting to happen.

Or did I miss the commit that actually uses `check` for something other than the error message?

Reply:
This is still silently accepted, I believe.

Reply:
yes